contributors/design-proposals/node/troubleshoot-running-pods.md

```diff
@@ -192,104 +225,86 @@ type EphemeralContainer struct {
 
-### Debug Container Status
+### Ephemeral Container Status
 
-The status of a Debug Container is reported in a new field in `v1.PodStatus`:
+`EphemeralContainer` is included in a new field in `PodStatus`:
 
 ```
 type PodStatus struct {
         ...
-        DebugStatuses []DebugStatus
-}
+        // List of user-initiated ephemeral containers that have been run in this pod.
+        // +optional
+        EphemeralContainers []EphemeralContainer `json:"commands,omitempty" protobuf:"bytes,11,rep,name=ephemeralContainers"`
 
-type DebugStatus struct {
-        Name string
-        Command []string
-        Args []string
-        // Set only for Debug Containers
-        DebugContainerStatus v1.ContainerStatus
 }
 ```
 
-Initially this will be populated only for Debug Containers, but there's interest
-in tracking status for traditional exec in a similar manner. Ideally we can
-report both types of user intervention into a container with a single new type.
-
-Note that `Command` and `Args` must be tracked in the status object because
-there is no spec for Debug Containers or exec. These must either be made
-available by the runtime or tracked by the kubelet. For Debug Containers this
-could be stored as runtime labels, but the kubelet currently has no method of
-storing state across restarts for exec. Solving this problem for exec is out of
-scope for Debug Containers, but we will look for a solution as we implement this
-feature.
-
-`DebugStatuses` is populated by the kubelet in the same way as regular and init
+The kubelet should be able to construct a complete `PodStatus` with no prior
+state using information stored in the container runtime.
+`EphemeralContainer.Status` introduces no new data, but the kubelet must also
+now populate `EphemeralContanier.Spec` &
+`EphemeralContainer.TargetContainerName`.
+
+The kubelet already persists container metadata as CRI
+[labels](https://github.com/kubernetes/kubernetes/blob/v1.10.0-alpha.0/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto#L606)
+and
+[annotations](https://github.com/kubernetes/kubernetes/blob/v1.10.0-alpha.0/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto#L613).
+The entire v1.Container used in the request will be serialized and stored as a
+runtime annotation. The value of `TargetContainerName` will be stored as a
+runtime label. Persisting this data in the runtime means it will survive kubelet
+restarts.
+
+At least for the Docker runtime, this is [an intended use of docker
+labels](https://docs.docker.com/engine/userguide/labels-custom-metadata/#value-guidelines).
+Docker does not document the maximum length of labels in its API. Empirically,
+it supports up to the 64K constraint of the docker client's `bufio.Scanner`
+size. Because the container spec may be examined in security sensitive contexts
+like admission control, we will conservatively limit the size of the spec to 32K
+and add a 32K minimum label length test to runtime qualification.
```

Comment:
@mrunalp This is the section where I'd appreciate your review. I'm planning on storing the v1.Container in the runtime. This works for Docker. Do you know of any limitations for cri-o?